test/e2e/httpproxy/multiple_ingress_classes_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build e2e
+// +build e2e
+
+package httpproxy
+
+import (
+	. "github.com/onsi/ginkgo"
+	contourv1 "github.com/projectcontour/contour/apis/projectcontour/v1"
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func testMultipleIngressClassesField(namespace string) {
+	Specify("multiple ingress class names with a single Contour instance (ingressClassName field)", func() {
+		t := f.T()
+
+		f.Fixtures.Echo.Deploy(namespace, "ingress-conformance-echo")
+
+		for _, class := range []string{"", "contour", "team1"} {
```

Comment:
An empty ingress class for the HTTPProxy wouldn't be valid here since Contour has been explicitly configured to process specific ingress classes.

Reply:
Thanks @skriss ! Such a simple fix but I couldn't see it.